src/features/users/usersSlice.ts

```diff
@@ -0,0 +1,50 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { getUsers } from '../../api/users';
+import { User } from '../../types/User';
+
+export const fetchUsers = createAsyncThunk('users/fetchUsers', async () => {
+  const response = await getUsers();
+
+  return response;
+});
+
+export interface UsersState {
+  users: User[];
+  loading: boolean;
+  error: string | null;
+}
+
+const initialState: UsersState = {
+  users: [],
+  loading: false,
+  error: null,
+};
+
+const usersSlice = createSlice({
+  name: 'users',
+  initialState,
+  reducers: {
+    resetUsers: state => {
+      state.users = [];
+      state.loading = false;
+      state.error = null;
+    },
+  },
+  extraReducers: builder => {
+    builder
+      .addCase(fetchUsers.pending, state => {
+        state.loading = true;
+      })
+      .addCase(fetchUsers.fulfilled, (state, action) => {
+        state.loading = false;
+        state.users = action.payload;
+      })
+      .addCase(fetchUsers.rejected, state => {
+        state.loading = false;
+        state.error = 'Failed to load users';
+      });
+  },
+});
+
+export default usersSlice.reducer;
+export const usersReducer = usersSlice.reducer;
```

Comment:
There are two exports for the reducer: `export default usersSlice.reducer;` and `export const usersReducer = usersSlice.reducer;`. Choose one export method to avoid redundancy. If the reducer is used in multiple places, ensure consistency in how it's imported.